### ISSUE TITLE
no such command: 'terminal'

### ISSUE THREAD

**Issue by BrianD91** [mawww/kakoune#3409]
### Steps 

install kakoune v2020.01.16 on WSL 2
install the fzf.kak plug-in
run `:fzf-mode` and press `f` in kakoune

### Outcome 

the following error is reported:

`Error: 1:2: 'fzf-file' 1:2: 'evaluate-commands' 4:1: 'fzf' 1:2:  'evaluate-commands' no such command: 'terminal'

### Expected 

fzf would open and operate correctly as the plug intends.

**Comment by Screwtapello** [comment]
Can you run the `:terminal` command interactively? How about `:new` or `:grep` or other commands in Kakoune's standard library?

**Comment by Screwtapello** [comment]
Actually now that I think about it - Kakoune currently supports five different implementations of the `:terminal`: command: tmux, screen, x11 (which is mostly just "execute a particular command to open a new terminal", not x11-specific), kitty (a cross-platform terminal emulator) and iterm (a macOS terminal emulator). Each of those implementations carefully checks to see if they're appropriate before installing their particular implementation as the default.

Probably none of them are appropriate on WSL.

What terminal are you using for WSL? How can you programmatically open a new terminal running a given command?

**Comment by BrianD91** [comment]
When I run `:new`, I get the following:

`The 'terminal' alias must be defined to use this command`

the `:grep` command seems to work as expected.

I'm primarily using wsltty for doing stuff with WSL2: https://github.com/mintty/wsltty
But I can also open up PowerShell and run `ubuntu.exe` which will put me into a bash terminal at my `$HOME` directory

**Comment by caksoylar** [comment]
You would have two options on WSL, either use an X server like [vcxsrv](https://sourceforge.net/projects/vcxsrv/) and a terminal running on X (instead of wsltty), or use a terminal multiplexer like [tmux](https://github.com/tmux/tmux/wiki). Tmux has a learning curve but it is what I use with wsltty. It comes very handy for session management and working with multiple terminal windows.

**Comment by mawww** [comment]
`fzf.kak` needs a way to open a new terminal to display fzf in, it does indeed look like none of the current providers for the `terminal` alias are available on your WSL setup.

It would be nice to add support for that, probably through something similar to `x11.kak` that would be named `wsl.kak` and would:

* check somehow that we are in a WSL environment
* if that is the case, loop through known commands that can start a terminal on WSL (checking if `wsltty` is runable, checking if whatever the default provided terminal is there...)
* set `termcmd` to whatever best terminal command was found (it could share the `termcmd` option with `x11.kak`, or could introduce its own).

**Comment by Screwtapello** [comment]
Looking at [the `wsltty` docs](https://github.com/mintty/wsltty#command-line-scripts-wslbat), it seems like you should be able to do the thing you want by convincing Kakoune to launch `WSL.bat` to open a new terminal... but somebody will have to figure out the logistics of hooking it up as a `:wsl-terminal` command that can be aliased to `:terminal`.

**Comment by andreyorst** [comment]
> or use a terminal multiplexer like `tmux`

I second `tmux` solution as well, because Kakoune has extremely good support for tmux out of the box, and there are many plugins that support it as well. At least most of my plugins do.

Kakoune is focused on POSIX environment, and while Windows10 is not in the scope of Kakoune's focus, I still think `wsl-terminal` should exist in form of a plugin or internally.

**Comment by BrianD91** [comment]
well I installed tmux in my ubuntu build on WSL2, and it seems to fix THAT problem! 🎉 

But, now two more problems have arisen >_>

First, there's a tremendous lag when switching modes. If I go to insert mode and type some inputs then press esc to leave insert mode it takes nearly a full second for it to actually switch modes. and if I enter any input before the mode successfully switches it will assume those should be inserted into the file as it's still in insert mode and it will cancel out switching out of insert mode. Sometimes it wouldn't cancel at all until I pressed Esc several times.

Second, the colors don't seem to be working appropriately. I tried `tmux -2` and other config type stuff but could seem to get kakoune to actually render the intended colors.
